### PR TITLE
Log details when extension assembly types can't be loaded.

### PR DESCRIPTION
**Pull request by barnson**

### CODE REVIEW

File: src/WixToolset.BuildTasks/DoIt.cs [hunk 1]
```diff
@@ -170,13 +172,13 @@ private void ExecuteCore()
...
             var commandLine = serviceProvider.GetService<ICommandLineParser>();
-            commandLine.ExtensionManager = this.CreateExtensionManagerWithStandardBackends(serviceProvider, arguments.Extensions);
+            commandLine.ExtensionManager = this.CreateExtensionManagerWithStandardBackends(serviceProvider, messaging, arguments.Extensions);
             commandLine.Arguments = arguments;
             var command = commandLine.ParseStandardCommandLine();
             command?.Execute();
         }
 
-        private IExtensionManager CreateExtensionManagerWithStandardBackends(IServiceProvider serviceProvider, string[] extensions)
+        private IExtensionManager CreateExtensionManagerWithStandardBackends(IServiceProvider serviceProvider, IMessaging messaging, string[] extensions)
```
Comment (rseanhall) [inline src/WixToolset.BuildTasks/DoIt.cs]: I'm slowly reading through all the commits that were done in the last year. I see that this code is still duplicated in wix/Program.cs. Why? I think there should be something like `IExtensionManagerFactory` that does this.

Comment (barnson) [reply src/WixToolset.BuildTasks/DoIt.cs]: And in Light. Avoiding duplication is a good. Avoiding death by a thousand little factories is also a good. Remember, if we get to the point of having any kind of `FactoryFactory`, we all have to give up hope. :)

Comment (barnson) [reply src/WixToolset.BuildTasks/DoIt.cs]: That's my take, anyway. I felt dirty duplicating the `catch`, I admit. @robmen?

Comment (robmen) [reply src/WixToolset.BuildTasks/DoIt.cs]: This is a particularly special case. This code is what ties the "tools" to the supported (aka: built-in, aka: standard) backends. The `WixToolset.Core` assembly should not have dependencies on the backend assemblies so there isn't any other natural place for this code to live (without creating yet another assembly that has only this block of code in them).

Comment (rseanhall) [reply src/WixToolset.BuildTasks/DoIt.cs]: I felt the need for a new WixToolset.Tools.Core assembly in my WixCop pull request. It was pretty small, but this seems like there's going to be more and more need for such a thing.

**Pull request merged**